SpatialGDK/Source/SpatialGDKFunctionalTests/Private/TestMaps/Spatial2WorkerFullInterestMigrationMap.cpp

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "TestMaps/Spatial2WorkerFullInterestMigrationMap.h"
+#include "EngineClasses/SpatialWorldSettings.h"
+#include "GameFramework/PlayerStart.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/SpatialTestCharacterMigration/SpatialTestCharacterMigration.h"
+#include "SpatialGDKFunctionalTests/SpatialGDK/SpatialTestCharacterMovement/CharacterMovementTestGameMode.h"
+#include "TestWorkerSettings.h"
+
+USpatial2WorkerFullInterestMigrationMap::USpatial2WorkerFullInterestMigrationMap()
+	: UGeneratedTestMap(EMapCategory::CI_PREMERGE_SPATIAL_ONLY, TEXT("Spatial2WorkerFullInterestMigrationMap"))
+{
+	SetNumberOfClients(2);
+}
+
+void USpatial2WorkerFullInterestMigrationMap::CreateCustomContentForMap()
+{
+	ULevel* CurrentLevel = World->GetCurrentLevel();
+
+	// Add the tests
+	AddActorToLevel<ASpatialTestCharacterMigration>(CurrentLevel, FTransform::Identity);
+
+	FTransform PlayerTransform;
+
+	PlayerTransform.SetTranslation(FVector(-500.0f, -500.0f, 50.0f));
+	AddActorToLevel<APlayerStart>(CurrentLevel, PlayerTransform);
+
+	PlayerTransform.SetTranslation(FVector(-500.0f, 500.0f, 50.0f));
+	AddActorToLevel<APlayerStart>(CurrentLevel, PlayerTransform);
```

Comment:
feels like this could be refactored so the two maps descend from something common

Reply:
but it could be more work than it's worth. though you could combine the source files so it doesn't sprawl that many places.

Reply:
Have opted to add them into one cpp. Deriving from a common class has the issue of the common class being "generated" when generate maps. Perhaps there is a nice way around this?

Reply:
Ah, true - you can theoretically override `ShouldGenerateMap()` to control this, but it's probably more hassle than it's worth.
I think having it in the same file, possibly using a file-local function should be enough.

Reply:
You unfortunately AddActorToLevel is protected so I can't pull out a function.

Reply:
whoops